Zend/Optimizer/zend_func_info.c

```diff
@@ -100,18 +101,20 @@ ZEND_API int zend_func_info_rid = -1;
 
 uint32_t zend_get_internal_func_info(
 		const zend_function *callee_func, const zend_call_info *call_info, const zend_ssa *ssa) {
-	if (callee_func->common.scope) {
-		/* This is a method, not a function. */
+	if (callee_func->common.scope && !call_info->is_prototype) {
```

Comment:
Shouldn't this be `callee_func->common.scope && call_info->is_prototype`? `is_prototype` indicates we might be calling an overridden method.